stdlib/Random/test/runtests.jl

```diff
@@ -801,3 +801,29 @@ end
 @testset "RNGs broadcast as scalars: T" for T in (MersenneTwister, RandomDevice)
     @test length.(rand.(T(), 1:3)) == 1:3
 end
+
+const BASE_TEST_PATH = joinpath(Sys.BINDIR, "..", "share", "julia", "test")
+isdefined(Main, :OffsetArrays) || @eval Main include(joinpath($(BASE_TEST_PATH), "testhelpers", "OffsetArrays.jl"))
+
+@testset "OffsetArrays" begin
+    # axes as tuple
+    @test axes(rand(Bool, (Base.OneTo(3), Base.OneTo(3)))) === (Base.OneTo(3), Base.OneTo(3))
+    @test axes(rand(Bool, (Base.OneTo(3), 3))) === (Base.OneTo(3), Base.OneTo(3))
+    @test axes(rand(Bool, (0:2, 0:2))) == (0:2, 0:2)
+
+    # 1D axes
+    @test axes(rand(Bool, Base.OneTo(3))) === (Base.OneTo(3), )
+    @test axes(rand(Bool, 3)) === (Base.OneTo(3), )
+    @test axes(rand(Bool, 0:2)) == (0:2, )
+
+    # axes splatted
+    @test axes(rand(Bool, Base.OneTo(3), Base.OneTo(3))) === (Base.OneTo(3), Base.OneTo(3))
+    @test axes(rand(Bool, Base.OneTo(3), 3)) === (Base.OneTo(3), Base.OneTo(3))
+    @test axes(rand(Bool, 0:2, 0:2)) == (0:2, 0:2)
+
+    # edge cases
+    @test rand(3) isa Array{Float64, 1}
+    @test rand(1:3) isa Integer
+    @test rand(1:3, 1:3) isa OffsetArrays.OffsetArray
```

Comment:
What would happen if `OffsetArrays` isn't imported?

Reply:
That's the error I would expect, too. I agree it should be tested. I don't know how to test it since tests before this test runs could have loaded `OffsetArrays.jl`. Wrapping the test code in a new module would not hide the definition of `Base.similar`